go-controller/pkg/ovn/pods.go

```diff
@@ -291,8 +291,18 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) {
 		return
 	}
 
-	annotation := fmt.Sprintf(`{\"ip_address\":\"%s/%s\", \"mac_address\":\"%s\", \"gateway_ip\": \"%s\"}`,
-		podIP.String(), mask, podMac.String(), gatewayIP)
+	ip, ipnet, err := net.ParseCIDR(podIP.String() + "/" + mask)
+	if err != nil {
+		logrus.Errorf("error parsing pod IP %s/%s: %v", podIP, mask, err)
+		return
+	}
+	ipnet.IP = ip
+	annotation, err := util.MarshalPodAnnotation(ipnet, podMac, gatewayIP, nil)
```

Comment:
Given that `MarshalPodAnnotation` just wants a string, why not just pass `podIP.String() + "/" + mask` to it directly rather than parsing and unparsing it first?